File: ru.kmorozov.api.onedrive/src/main/java/com/kmorozov/onedrive/client/OneDriveAPIException.java
package com.kmorozov.onedrive.client;

import java.io.IOException;

public class OneDriveAPIException extends IOException {

    private final int code;

    public OneDriveAPIException(final int code, final String message) {
        super(message);
        this.code = code;
    }

    public OneDriveAPIException(final int code, final String message, final Throwable cause) {
        super(message, cause);
        this.code = code;
    }

    public int getCode() {
        return code;
    }
}

File: ru.kmorozov.api.onedrive/src/main/java/ru/kmorozov/onedrive/client/OneDriveAPIException.kt
package ru.kmorozov.onedrive.client

import java.io.IOException

class OneDriveAPIException : IOException {

    val code: Int

    constructor(code: Int, message: String) : super(message) {
        this.code = code
    }

    constructor(code: Int, message: String, cause: Throwable) : super(message, cause) {
        this.code = code
    }
}
